pkg/devicemapper/devmapper.go

```diff
@@ -342,7 +341,7 @@ func RemoveDevice(name string) error {
 		return fmt.Errorf("devicemapper: Error running RemoveDevice %s", err)
 	}
 
-	return nil
+	return UdevWait(cookie)
```

Comment:
Capturing error code from UdevWait() makes sense. I think you could still use defer UdevWait() but to capture error you will have to use named error variable and in deferred function set that named error variable from return of UdeWait(). That means this will require a wrapper around UdevWait().

Docker developers typically like to use "defer" instead over inline calls. But I am fine with either.